Integrations/integration-ProtectWise.yml

```diff
@@ -201,101 +187,300 @@ script:
         case 'event-pcap-info':
-            var token = getToken();
-            return {Type: 1, Contents: getEventPcapInfo(args.eventId,token), ContentsFormat: 'json'};
+        case 'protectwise-event-pcap-info':
+            var url = params.url+'pcaps/events/'+ args.eventId + '/info';
+            var res = doReq(url, token);
+            var md = '### Protectwise PCAP for Event ' + res.id + '\n - Estimated size: ' + res.estimatedSize + ' bytes';
+            if (res.netflows && res.netflows.length > 0) {
+                var nf = filterClone(res.netflows, DEFAULT_NETFLOW_HEADERS );
+                md += '\n' + tableToMarkdown('Included Netflows', res.netflows);
+            }
+
+            return {
+                Type: entryTypes.note,
+                Contents: res,
+                ContentsFormat: formats.json,
+                HumanReadable: md,
+                EntryContext: {'Protectwise.Event(val.id == obj.id)': {'id': res.id, 'PCAPSize': res.estimatedSize}}
+            };
         case 'observation-pcap-download':
-            var token = getToken();
-            return {Type: 3, FileID: observationEventPcap(args.id,args.sensorId,args.filename,token), File: args.filename, Contents: 'we must have contents for an entry'};
+        case 'protectwise-observation-pcap-download':
+            var filename = (args.filename && args.filename.length > 0) ? args.filename : ( args.sensorId+'-'+args.id + '.pcap');
+            var url = params.url+'pcaps/observations/'+args.sensorId+'/'+args.id + '?filename=' + filename;
+            res = saveFile(doReq(url,token,true));
+            return {Type: 3, FileID: res, File: filename, Contents: 'we must have contents for an entry'};
         case 'observation-pcap-info':
-            var token = getToken();
-            return {Type: 1, Contents: getObservationPcapInfo(args.id,args.sensorId,token), ContentsFormat: 'json'};
+        case 'protectwise-observation-pcap-info':
+            var url = params.url+'pcaps/observations/'+ args.sensorId + '/' + args.id + '/info';
+            var res = doReq(url, token);
+            var md = '### Protectwise PCAP for Observation ' + res.id + '\n - Estimated size: ' + res.estimatedSize + ' bytes';
+            if (res.netflows && res.netflows.length > 0) {
+                var nf = filterClone(res.netflows, DEFAULT_NETFLOW_HEADERS );
+                md += '\n' + tableToMarkdown('Included Netflows', res.netflows);
+            }
+
+            return {
+                Type: entryTypes.note,
+                Contents: res,
+                ContentsFormat: formats.json,
+                HumanReadable: md,
+                EntryContext: {'Protectwise.Observation(val.id == obj.id && val.sensorId == obj.sensorId)': {'id': args.id, 'sensorId': args.sensorId, 'PCAPSize': res.estimatedSize}}
+            };
         case 'get-token':
-            return getToken();
+            return token;
```

Comment:
getToken always gets called, before checking which command it is - so no reason to call it again, just return the token we already got